apps/opik-frontend/src/components/pages/PlaygroundPage/PlaygroundOutputs/PlaygroundOutputScores/PlaygroundOutputScores.tsx

```diff
@@ -0,0 +1,341 @@
+import React, { useMemo, useRef, useEffect } from "react";
+import { Loader2 } from "lucide-react";
+
+import useTraceById from "@/api/traces/useTraceById";
+import useRulesList from "@/api/automations/useRulesList";
+import useProjectByName from "@/api/projects/useProjectByName";
+import useAppStore from "@/store/AppStore";
+import FeedbackScoreTag from "@/components/shared/FeedbackScoreTag/FeedbackScoreTag";
+import {
+  HoverCard,
+  HoverCardContent,
+  HoverCardTrigger,
+} from "@/components/ui/hover-card";
+import { cn } from "@/lib/utils";
+import { TraceFeedbackScore } from "@/types/traces";
+import { PLAYGROUND_PROJECT_NAME } from "@/constants/shared";
+import { TAG_VARIANTS_COLOR_MAP } from "@/components/ui/tag";
+import { generateTagVariant } from "@/lib/traces";
+import { EvaluatorsRule, EVALUATORS_RULE_TYPE } from "@/types/automations";
+
+const MAX_VISIBLE_METRICS = 3;
+
+// Helper to get score names that a rule will produce
+const getScoreNamesFromRule = (rule: EvaluatorsRule): string[] => {
+  if (
+    rule.type === EVALUATORS_RULE_TYPE.llm_judge ||
+    rule.type === EVALUATORS_RULE_TYPE.thread_llm_judge ||
+    rule.type === EVALUATORS_RULE_TYPE.span_llm_judge
+  ) {
+    return rule.code.schema?.map((s) => s.name) || [];
+  }
+  if (
+    rule.type === EVALUATORS_RULE_TYPE.python_code ||
+    rule.type === EVALUATORS_RULE_TYPE.thread_python_code ||
+    rule.type === EVALUATORS_RULE_TYPE.span_python_code
+  ) {
+    return rule.code.metric ? [rule.code.metric] : [];
+  }
+  return [];
+};
+
+interface PlaygroundOutputScoresProps {
+  traceId: string | null;
+  selectedRuleIds: string[] | null | undefined;
+  stale?: boolean;
+  className?: string;
+  outputReady?: boolean;
+}
+
+const REFETCH_INTERVAL = 5000; // 5 seconds
+const MAX_REFETCH_TIME = 300000; // 5 minutes max polling
+
+const PlaygroundOutputScores: React.FC<PlaygroundOutputScoresProps> = ({
+  traceId,
+  selectedRuleIds,
+  stale = false,
+  className,
+  outputReady = false,
+}) => {
+  const workspaceName = useAppStore((state) => state.activeWorkspaceName);
+
+  // Track when polling started for timeout calculation
+  const pollingStartTimeRef = useRef<number | null>(null);
+
+  // Reset polling start time when traceId changes
+  useEffect(() => {
+    pollingStartTimeRef.current = traceId ? Date.now() : null;
+  }, [traceId]);
+
+  // Rules are selected if:
+  // - selectedRuleIds is undefined (legacy outputs without stored selection - treat as "all")
+  // - selectedRuleIds is null (explicitly "all" selected)
+  // - selectedRuleIds is a non-empty array (specific rules selected)
+  const selectingAllRules =
```

Comment:
pre-commit should be added to `.git/.hooks` folder, in root directory

See scripts:
1. hooks-install.sh
2. hooks-remove.sh